From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Fri, 02 Oct 2026 02:30:34 +0000
Subject: [PATCH] Fenix - IronFox Settings: Firefox VPN

Provides support for a UI setting to enable/disable Firefox VPN (IP Protection).

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/ipprotection/FenixIPProtectionEligibilityStorage.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/ipprotection/FenixIPProtectionEligibilityStorage.kt
index 212e6c60df3c..41774aa13ca0 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/ipprotection/FenixIPProtectionEligibilityStorage.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/ipprotection/FenixIPProtectionEligibilityStorage.kt
@@ -18,6 +18,9 @@ import mozilla.components.feature.ipprotection.IPProtectionEligibilityStorage
 import mozilla.components.feature.ipprotection.store.state.EligibilityStatus
 import org.mozilla.fenix.nimbus.FxNimbus
 import org.mozilla.fenix.settings.registerOnSharedPreferenceChangeListener
+import android.content.Context
+import org.ironfoxoss.ironfox.utils.IronFoxPreferences
+import org.mozilla.fenix.ext.components
 
 /**
  * Fenix's implementation of [IPProtectionEligibilityStorage].
@@ -32,6 +35,7 @@ class FenixIPProtectionEligibilityStorage(
     private val sharedPref: SharedPreferences,
     private val prefKey: String,
     private val lifecycleOwner: LifecycleOwner,
+    private val context: Context,
 ) : IPProtectionEligibilityStorage {
 
     private val secretEnabled = MutableStateFlow(sharedPref.getBoolean(prefKey, false))
@@ -42,10 +46,13 @@ class FenixIPProtectionEligibilityStorage(
                 secretEnabled,
             ) { region, secretOverride ->
                 val nimbus = FxNimbus.features.ipProtection.value()
+                val fxaEnabled = IronFoxPreferences.isFxaEnabled(context)
+                val fxaSignedIn = context.components.settings.signedInFxaAccount
+                val ipProtectionEnabled = IronFoxPreferences.isIPProtectionEnabled(context)
                 val status =
                     when {
                         secretOverride -> EligibilityStatus.Eligible
-                        !nimbus.enabled -> EligibilityStatus.Ineligible
+                        !ipProtectionEnabled || !fxaEnabled || !fxaSignedIn || !nimbus.enabled -> EligibilityStatus.Ineligible
                         region?.home in nimbus.allowedRegions -> EligibilityStatus.Eligible
                         else -> EligibilityStatus.UnsupportedRegion
                     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/ipprotection/IPProtection.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/ipprotection/IPProtection.kt
index a2f3b58abf0c4..e34c3333f5d82 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/ipprotection/IPProtection.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/ipprotection/IPProtection.kt
@@ -70,6 +70,7 @@ class IPProtection(
             sharedPref = settings.preferences,
             prefKey = context.getString(R.string.pref_key_enable_ip_protection),
             lifecycleOwner = ProcessLifecycleOwner.get(),
+            context = context,
         )
     }
 
-- 
